sound/soc/sof/loader.c

```diff
@@ -873,6 +873,15 @@ int snd_sof_run_firmware(struct snd_sof_dev *sdev)
 		return ret;
 	}
 
+	mutex_lock(&sdev->cores_status_mutex);
+	/*
+	 * fw booted, Only the master DSP core (0) should be ON here,
+	 * update to reflect it.
+	 */
+	sdev->core_refs[0] = 1;
```

Comment:
I think we should tie the core_ref incrementing/decrementing to the core_power_up()/power_down() instead of hardcoding it. We've run into problems with this in the past before we cleaned up the enabled_cores_mask to be updated in the power_up/power_down ops